File: src/main/java/com/dansoftware/libraryapp/exception/UncaughtExceptionHandler.java
package com.dansoftware.libraryapp.exception;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class UncaughtExceptionHandler implements Thread.UncaughtExceptionHandler {

    private static final Logger LOGGER = LoggerFactory.getLogger(UncaughtExceptionHandler.class);

    @Override
    public void uncaughtException(Thread t, Throwable e) {
        LOGGER.error("Uncaught exception occurred", e);
    }
}

File: src/main/java/com/dansoftware/libraryapp/exception/UncaughtExceptionHandler.kt
package com.dansoftware.libraryapp.exception

import org.slf4j.Logger
import org.slf4j.LoggerFactory

class UncaughtExceptionHandler : Thread.UncaughtExceptionHandler {
    override fun uncaughtException(t: Thread, e: Throwable) {
        LOGGER.error("Uncaught exception occurred", e)
    }

    companion object {
        private val LOGGER: Logger = LoggerFactory.getLogger(UncaughtExceptionHandler::class.java)
    }
}